apps/explorer/src/ui/DateCard.tsx

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Text } from '~/ui/Text';
+import { formatDate } from '~/utils/timeUtils';
+
+export interface DateCardProps {
+    date: Date | number;
+}
+
+// TODO - add format options
+export function DateCard({ date }: DateCardProps) {
+    const dateStr = formatDate(date, [
+        'month',
+        'day',
+        'year',
+        'hour',
+        'minute',
+    ]);
+    return dateStr && dateStr !== '' ? (
```

Comment:
Prefer early return instead of ternaries when we can.

```
if (!dateStr) {
  return null;
}

return <div>
```

Reply:
Although it seems like this really shouldn't be possible given the type of the date prop? I feel like this should throw an error maybe?